### PR TITLE
[SYCLomatic] Support the migration of 4 CUB api

### PR DESCRIPTION
**Pull request by yihwang-nv**
- cub::DeviceScan::InclusiveScanByKey
- cub::DeviceScan::InclusiveSumByKey
- cub::DeviceScan::ExclusiveScanByKey
- cub::DeviceScan::ExclsuvieSumByKey

Signed-off-by: Wang, Yihan <yihan.wang@intel.com>

### CODE REVIEW

File: clang/lib/DPCT/CallExprRewriter.h [hunk 1]
```diff
@@ -798,6 +797,18 @@ template <class NameT, class... TemplateArgsT> class TemplatedNamePrinter {
   }
 };
 
+/// Print a `type`. This is useful when print no template type.
```
Comment (AndyCHHuang) [inline clang/lib/DPCT/CallExprRewriter.h]: Print a type with no template.

**Review by AndyCHHuang — CHANGES_REQUESTED** [review]

File: clang/lib/DPCT/CallExprRewriter.h (hunk 1 above)
Comment (yihwang-nv) [reply clang/lib/DPCT/CallExprRewriter.h]: done

**Review by AndyCHHuang — APPROVED** [review]
LGTM

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**